veps/sig-network/passt-migration-proposal.md

```diff
@@ -0,0 +1,284 @@
+        devices:
+          interfaces:
+          - name: passtnet
+            binding:
+              name: passt
+            ports:
+            - name: http
+              port: 80
+              protocol: TCP
+```
+
+## Alternatives
+
+TCP live migration requires a privileged action (TCP_REPAIR socket option). A straightforward approach is granting passt the required capability (`CAP_NET_ADMIN`). Ambient capabilities can be applied to the passt binary similarly to [virt-launcher](https://github.com/kubevirt/kubevirt/blob/release-1.5/cmd/virt-launcher-monitor/virt-launcher-monitor.go#L176).
+
+However, container runtimes drop all capabilities from the [bounding set](https://man7.org/linux/man-pages/man7/capabilities.7.html), and capabilities must explicitly be added in the SecurityContext, as done for CAP_NET_BIND_SERVICE in [virt-launcher](https://github.com/kubevirt/kubevirt/blob/release-1.5/pkg/virt-controller/services/rendercontainer.go#L284).
+
+Since virt-launcher drops the root user, this method is secure. However, KubeVirt follows Kubernetes' `Restricted` Pod Security Standard, which disallows CAP_NET_ADMIN.
+
+## Scalability
+
+- `passt` has a memory overhead of ~250Mi per VM. Users running VMs at scale should revert to other bindings if memory overhead is a concern.
+-  VMs with a large number of TCP connections will take longer to live migrate. That's because sockets are handled in a serial
+ fashion, each one separately sent back and forth to passt-repair for handling.
+- The number of concurrent live migration per node is already limited and bound by the number of virt-handler worker goroutines.
+
+## Update/Rollback Compatibility
+
+- Live migration of passt VMs from nodes running older versions of KubeVirt to an upgraded KubeVirt environment will result in TCP connections being reset.
+- Rollback of passt migration functionality (in case system stability is impacted) will be possible by disabling of the feature gate. It is expected that passt would still work, but live migrations would disconnect TCP connections.
+
+## Functional Testing
+
+Existing e2e tests cover basic passt connectivity and remain sufficient with vhost-user support. Dedicated e2e tests for seamless migration will verify TCP connection persistence. Sticky IP addresses are required, so tests will use secondary interfaces with hardcoded IPs.
+
+## Implementation Phases
+
+1. Implement vhost-user in passt-sidecar.
+2. Add passt-repair to virt-handler.
+3. Implement calls to passt-repair in virt-handler.
+4. Enhance e2e tests and demonstrate migration without disconnection.
+5. Deployment activities.
+
+Phases can proceed in parallel.
+
+## Feature Lifecycle Phases
+
+`passt` is currently a network binding plugin external to the KubeVirt core, and its use as a network binding is not subject to lifecycle phases. However, the calls to passt-repair from virt-handler impose a risk and should be protected by a feature gate and subject to lifecycle phases.
```

Comment:
If we are to be a bit more accurate, it is a plugin, however is dependent and coupled to the passt package which resided in the compute container. The plugin parts are not subject to kubevirt/kubevirt lifecycle directly, however the passt package and the new addition in virt-handler are subject to it.
Please make this clear here.

Reply:
For some reason, I can't reply directly to the comment below, so let me reply here:

> If you can, please clarify that at this point the migration will include seamless connectivity treatment by default with no option to disable it.

...actually, if you need an option to disable it for whatever reason, you can just *not* start passt-repair. Migration will succeed but connections will be reset.

Reply:
I understand, thanks.
I actually prefer things without too many options, so when we reach GA, all should work with no knobs or special flows. Easier to maintain and simpler to understand.